enterprise/server/remote_execution/execution_server/execution_server_test.go

```diff
@@ -618,27 +628,30 @@ func testExecuteAndPublishOperation(t *testing.T, test publishTest) {
 		require.Equal(t, codes.NotFound, gstatus.Code(err), "Error should be NotFound, but is %v", err)
 	}
 
-	// Should also be able to fetch the ExecuteResponse from cache. See field
-	// comment on Execution.execute_response_digest for notes on serialization
-	// format.
-	cachedExecuteResponse, err := execution.GetCachedExecuteResponse(ctx, env.GetActionCacheClient(), taskID)
-	require.NoError(t, err)
-	assert.Empty(t, cmp.Diff(expectedExecuteResponse, cachedExecuteResponse, protocmp.Transform()))
-
-	// Should also have recorded usage.
-	ut := env.GetUsageTracker().(*fakeUsageTracker)
-	var executionUsages []usage
-	for _, u := range ut.usages {
-		if u.labels.Client == "executor" && (u.counts.LinuxExecutionDurationUsec > 0 || u.counts.SelfHostedLinuxExecutionDurationUsec > 0) {
-			executionUsages = append(executionUsages, u)
+	// If retrying the execution, we don't cache the execution response or record usages.
+	if !test.retry {
```

Comment:
Instead of skipping the assertions in this case, could we do an assertion that the cached execute response will be NotFound (or nil) + assert that there is no usage tracked?